router/src/validation.rs

```diff
@@ -663,8 +662,23 @@ fn image_tokens(
                         max_longest_edge_for_image_resize,
                     )
                 }
-            } else {
-                (height, width)
+            };
+
+            let (height, width) = {
+                let aspect_ratio = height as f32 / width as f32;
+                if height >= width && height > max_image_size {
+                    (
+                        max_image_size,
+                        (max_image_size as f32 / aspect_ratio) as usize,
+                    )
+                } else if width > height && width > max_image_size {
+                    (
+                        (max_image_size as f32 * aspect_ratio) as usize,
+                        max_image_size,
+                    )
+                } else {
+                    (height, width)
+                }
```

Comment:
Why keep 2 loops ? Why not just change the value of 1456 to 4096 (Although I guess 1456 was done on purpose)

Reply:
the first loop is equal to https://github.com/huggingface/transformers/blob/main/src/transformers/models/idefics3/image_processing_idefics3.py#L139.     # Find the output size, when rescaling the longest edge to max_len and preserving the aspect ratio
the second loop is equal to https://github.com/huggingface/transformers/blob/main/src/transformers/models/idefics3/image_processing_idefics3.py#L141 #  # Find the output size when scaling the image to be below the MAX_IMAGE_SIZE